Tutorial Working with Tags (#226)
foundriesio/docs: source/tutorials/working-with-tags/adapting-shellhttpd.rst

	     echo "= $(date) ============================="
     done

Edit the file ``Dockerfile`` according to the example below:

.. prompt:: bash host:~$, auto

    host:~$ gedit shellhttpd/Dockerfile

**shellhttpd/Dockerfile**:

.. prompt:: text

     FROM alpine
     
     COPY httpd.sh /usr/local/bin/
     
     CMD ["/usr/local/bin/httpd.sh"]

Edit the file ``docker-compose.yml`` according to the example below:

.. prompt:: bash host:~$, auto

    host:~$ gedit shellhttpd/docker-compose.yml

**shellhttpd/docker-compose.yml**:

.. prompt:: text

     version: '3.2'
     
     services:
       httpd:
         image: hub.foundries.io/cavel/shellhttpd:latest
         restart: always
              ports:
           - 8080:${PORT-8080}
         environment:
           MSG: "Tag devel, test:01"

Note that ``MSG`` is defined with ``This is the TEST 01``.

Commit and push all changes done in the ``containers`` folder:

.. prompt:: bash host:~$, auto

    host:~$ git status
    host:~$ git add shellhttpd/docker-compose.yml
    host:~$ git add shellhttpd/httpd.sh
    host:~$ git add shellhttpd/Dockerfile
    host:~$ git commit -m "This is the TEST 02"
    host:~$ git push

> Make sure you received your update by checking the latest **Target** on the :guilabel:`Devices` tab 

Reviewer: I believe in this step we should suggest the user to wait for their build, is that correct?
```suggestion
Wait for your build to finish by checking the latest **Target** on the :guilabel:`Devices` tab 
```

     4        devel   shellhttpd                                             raspberrypi3-64
     5        devel   shellhttpd                                             raspberrypi3-64
     6        devel   shellhttpd                                             raspberrypi3-64
     7        devel   shellhttpd                                             raspberrypi3-64
     8        devel   shellhttpd-mqtt,mosquitto,shellhttpd,flask-mqtt-nginx  raspberrypi3-64
     9        devel   mosquitto,shellhttpd,flask-mqtt-nginx,shellhttpd-mqtt  raspberrypi3-64
     10       devel   mosquitto,shellhttpd,flask-mqtt-nginx,shellhttpd-mqtt  raspberrypi3-64

Check what **Target** your device is running:

.. prompt:: bash host:~$, auto

    host:~$ fioctl device list -f <factory>

**Example Output**:

.. prompt:: text

     NAME           FACTORY     TARGET                 STATUS  APPS                                        UP-TO-DATE
     ----           -------     ------                 ------  ----                                        ----------
     <device-name>  <factory>   raspberrypi3-64-lmp-10 OK      flask-mqtt-nginx,mosquitto,shellhttpd-mqtt  true

> If you keep changing the ``devel`` branch, FoundriesFactory CI will keep building 
> and generating **Targets** tagged with ``devel``. As a result, devices following 
> ``devel`` will keep updating to the latest **Target**.

Reviewer: ```suggestion
Whenever you change the ``devel`` branch, FoundriesFactory CI will build 
and generate a new **Target** tagged with ``devel``. As a result, devices following 
``devel`` will update to the latest **Target**.
```
